backend/Code/app.py

```diff
@@ -1,94 +1,68 @@
-            response = requests.get(ticker_to_CIK_URL, headers=headers)
-            soup = BeautifulSoup(response.content, "html5lib")
-            txt_dirty = soup.get_text()
-            # Clean the data
-            lines = [line.strip() for line in txt_dirty.split('\n')]
-            lines = [line.strip() for line in txt_dirty.split()]
-            ticker_cik_dict = dict(zip(lines[::2], lines[1::2]))
-            cik_ticker_df = pd.DataFrame(
-                list(ticker_cik_dict.items()), columns=["Ticker", "CIK"])
-            # Sort the data frame
-            cik_ticker_df = cik_ticker_df.sort_values("Ticker")
-            cik_ticker_df = cik_ticker_df.reset_index(drop=True)
-            # Export the data frame to an Excel file
-            cik_ticker_df.to_excel("./tickers.xlsx", index=False)
-            # Get the CIK number of the company
-            cik_number_required = cik_ticker_df["CIK"].loc[cik_ticker_df["Ticker"]
-                                                           == ticker].values
-        except requests.exceptions.RequestException:
-            # If there is an error connecting to the SEC website, print an error message and return None
-            print("Error: Unable to connect to https://www.sec.gov/include/ticker.txt.")
-            return None
-    try:
-        # Try to get the first CIK number from the list of CIK numbers
-        cik_number_required = cik_number_required[0]
-        return cik_number_required
-    except IndexError:
-        # If the ticker symbol is not found, print an error message and return None
-        print(f"Error: Ticker {ticker} not found.")
-        return 'None Existant ticker symbol'
-
-
-# Get the data from the json file
-def get_local_json_data_as_pandas(path, which_data):
-    if which_data == 'q':
-        quarter_df = pd.read_json(
-            path_or_buf=path, orient='split', compression='infer')
-        return quarter_df
-    elif which_data == 'a':
-        annual_df = pd.read_json(
-            path_or_buf=path, orient='split', compression='infer')
-        return annual_df
+from pathlib import Path
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
 
+def getCikNum(ticker, update_required):
+   # Check if the ticker symbol exists in the current system
+   if Path("./tickers.xlsx").exists() and not update_required:
+       # Read the Excel file containing the ticker symbols and their corresponding CIK numbers
+       cik_ticker_df = pd.read_excel("./tickers.xlsx")
+       try:
+           # Try to get the CIK number corresponding to the given ticker symbol
+           cik_number_required = cik_ticker_df["CIK"].loc[cik_ticker_df["Ticker"]
+                                                       == ticker].values
+       except IndexError:
+           # If the ticker symbol is not found, call the function again with update_required set to True
+           return getCikNum(ticker=ticker, update_required=True)
+   else:
```

Comment:
that is the point, the code checks if the file exists and if it does exist it checks in the local excel file if the ticker exists withing the excel file if it dosent it calls itself with the update=True. if update is true it gets the new ticker list from the web and checks if the cik number exists. it wont enter an infinite loop because of the if else statement written in lines 47 and 51 respectively